arkouda/categorical.py

```diff
@@ -268,20 +268,19 @@ def from_return_msg(repMsg):
         stringsMsg = f"{repParts[1]}+{repParts[2]}"
         parts = {
             "categories": Strings.from_return_msg(stringsMsg),
-            "codes": create_pdarray(repParts[3])
+            "codes": create_pdarray(repParts[3]),
+            "_akNAcode": create_pdarray(repParts[4])
         }
 
-        if len(repParts) > 3:
-            name = repParts[4].split()[1]
-            if ".permutation" in name:
-                parts["permutation"] = create_pdarray(repParts[4])
-            elif ".segments" in name:
-                parts["segments"] = create_pdarray(repParts[4])
-            else:
-                raise ValueError(f"Unknown field, {name}, found in Categorical.")
-
-        if len(repParts) == 4:
-            parts["segments"] = create_pdarray(repParts[5])
+        if len(repParts) > 5:
+            for i in range(5, len(repParts)):
+                name = repParts[i].split()[1]
+                if ".permutation" in name:
+                    parts["permutation"] = create_pdarray(repParts[i])
+                elif ".segments" in name:
+                    parts["segments"] = create_pdarray(repParts[i])
+                else:
```

Comment:
I think this only works if both `.permutation` and `.segments` exists and won't if only one does. I think to handle either existing, you need to do something similar to what was originally there

```python
if len > 4:
  # at least one exists

if len == 5:
   # both exist so handle the other one
```

I think this original code wasn't super obvious, so it might be worth adding some comments explaining

Reply:
ohhhh i think i misread this! Nevermind this looks perfect!